q2_feature_table/plugin_setup.py

```diff
@@ -19,41 +19,47 @@
     package='q2_feature_table'
 )
 
-# TODO create decorator for promoting functions to methods. This info would
-# be moved to the decorator calls.
 plugin.methods.register_function(
     function=q2_feature_table.rarefy,
     # TODO use more restrictive primitive type for `depth`
     inputs={'table': FeatureTable[Frequency]},
-    parameters={'depth': Int},
-    outputs=[('rarefied_table', FeatureTable[Frequency])],
+    parameters={'counts_per_sample': Int},
+    outputs=[('rarefied_table',
+              FeatureTable[Frequency] % Properties('uniform_sampling'))],
```

Comment:
What do you think about using hyphens as a naming convention for properties? `'uniform-sampling'`

Not sure if there are reasons to avoid using hyphens. @ebolyen what do you think?